src/opentelemetry.erl

```diff
@@ -106,12 +112,42 @@
 set_default_tracer(Tracer) ->
     persistent_term:put({?MODULE, default_tracer}, Tracer).
 
+set_default_context_manager(ContextModule) ->
+    persistent_term:put({?MODULE, context_manager}, ContextModule).
+
+get_context_manager() ->
+    persistent_term:get({?MODULE, context_manager}, {ot_ctx_noop, []}).
+
 get_tracer() ->
     persistent_term:get({?MODULE, default_tracer}, {ot_tracer_noop, []}).
 
 get_tracer(_Name) ->
     persistent_term:get({?MODULE, default_tracer}, {ot_tracer_noop, []}).
 
+set_http_extractor(List) ->
+    Fun = fun(Headers) ->
+              lists:foldl(fun(Extract, ok) ->
+                              Extract(Headers),
+                              ok
+                          end, ok, List),
+              ok
+          end,
+    persistent_term:put({?MODULE, http_extractor}, Fun).
```

Comment:
This module is no longer safe for hot-code loading, since after two editions of this module being loaded, the Fun will need to be reaped, ending up killing all processes that hold a reference to it?

Though from a quick test, it seems like it just eventually turns into a badfun, so it's not that bad?

I guess the only workaround would be to extract the function to have a kind of "state" argument and use a fully qualified fun to make it work. Probably not worth the effort at this point in time?